contracts/token/ERC20/ERC20Snapshot.sol

```diff
@@ -0,0 +1,130 @@
+pragma solidity ^0.4.24;
+
+import "./ERC20.sol";
+import "../../utils/Arrays.sol";
+
+
+/**
+ * @title ERC20Snapshot token
+ * @dev An ERC20 token which enables taking snapshots of account balances.
+ * This can be useful to safely implement voting weighed by balance.
+ */
+contract ERC20Snapshot is ERC20 {
```

Comment:
There is no tracking of the total supply history of the token if it was mintable or burnable

Reply:
I don't see the reason to keep racking total supply. Could you elaborate ?

Reply:
Voting power and dividend payments require knowing the proper ratio of tokens held against the current token supply.

Reply:
Small typo in the comment here, should be "weighted" by balance.